### ISSUE TITLE
Unhandled exception

### ISSUE THREAD

**Issue by Frigyes06** [makeasnek/FindTheMag2#23]
```Starting control of BOINC...
Traceback (most recent call last):
  File "D:\FindTheMag2\main.py", line 2645, in <module>
    verification_result = loop.run_until_complete(verify_boinc_connection(rpc_client))
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "D:\FindTheMag2\main.py", line 1520, in verify_boinc_connection
    response = await rpc_client._request(req)
  File "D:\FindTheMag2\libs\pyboinc\rpc_client.py", line 78, in _request
    return await self._raw_client.request(req)
  File "D:\FindTheMag2\libs\pyboinc\_raw_client.py", line 82, in request
    return await self.receive()
  File "D:\FindTheMag2\libs\pyboinc\_raw_client.py", line 63, in receive
    buff = [await self._read()]
  File "D:\FindTheMag2\libs\pyboinc\_raw_client.py", line 57, in _read
    return await self._reader.readuntil(separator=END_OF_MESSAGE)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\asyncio\streams.py", line 613, in readuntil
    raise exceptions.IncompleteReadError(chunk, None)
asyncio.exceptions.IncompleteReadError: 0 bytes read on a total of undefined expected bytes
```

[debug.log](https://github.com/makeasnek/FindTheMag2/files/11715397/debug.log)

**Comment by HeyMerlin** [comment]
Hi @Frigyes06 
Confirming your setup:
- It looks like you do not have the Gridcoin wallet running on the machine that FTM2 is running, correct?
- It looks like FTM2 was unable to connect to the BOINC client. Is the BOINC client running on the same machine as FTM2?
- Have you customized the config.py file at all, or is it the default?

@makeasnek 
It looks like there should be a try/catch around the `response = await rpc_client._request(req)` lines to handle if there is a failed connection. Lines 1520, 1526, 1529, 1533, 743, 1277, 1292, 1300, 1308, 1400, 1405, 1479, 1484, 1494, 1503, and 1518.
Thoughts?

Also it passed `authorize_response = await rpc_client.authorize()` (line 1518) but not  `response = await rpc_client._request(req)` (line 1520 where the exception occurred). 
I need to learn the pyboinc lib to understand why. Care to enlighten me?

**Comment by Frigyes06** [comment]
I have both BOINC and GRC wallet running on the same machine as FTM2. I changed the GRC wallet path, as I have it in a non-standard place.

There definitely should be try-excepts around these requests.

**Comment by HeyMerlin** [comment]
Not being able to communicate with the Gridcoin wallet is okay. But I'm not sure why FTM could not communicate with the BOINC client. I assume you have remote connections configured in the BOINC client? That is: a password in your `gui_rpc_auth.cfg` file and that the file is readable by FTM?
FTM should have detected if this was not the case.

**Comment by Frigyes06** [comment]
Update: It now talks to my GRC wallet. Password is present in the rpc conf file. I set it in the FTM config to try, no avail. Same error

**Comment by HeyMerlin** [comment]
Is BOINC installed in the default location on drive C, or is it installed on drive D?
Not sure if it will make a difference, just trying to understand your setup and configuration.

**Comment by Frigyes06** [comment]
It's installed in drive D

**Comment by Frigyes06** [comment]
Any progress? Anything you need from me? I tried now without account manager, still no bueno

**Comment by HeyMerlin** [comment]
Sorry, I have not had a chance to look into this further. 

The one thing that you could do that may help is in the `config.py` file, set the logging level to debug (`log_level='DEBUG'`) instead of warning, and then reproduce the error message. 
Don't post the debug.log file as (@makeasnek) usually has users email the log file to him. I have not confirmed what private info gets logged so for now either post the part of the log around the crash. 

Alternatively you could email the log file to me (hey dot merlin dot hansen at gmail.com), but since I'm not an official contributor to the project yet you may not be comfortable with that.

I currently have FTM running on my Linux boxes but not my Windows boxes. I'll get it installed on a Windows box and see if I can reproduce the issue so I can better troubleshoot it.

**Comment by Frigyes06** [comment]
Error:
```
...
Starting control of BOINC...
Traceback (most recent call last):
  File "D:\FindTheMag2\main.py", line 2645, in <module>
    verification_result = loop.run_until_complete(verify_boinc_connection(rpc_client))
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "D:\FindTheMag2\main.py", line 1520, in verify_boinc_connection
    response = await rpc_client._request(req)
  File "D:\FindTheMag2\libs\pyboinc\rpc_client.py", line 78, in _request
    return await self._raw_client.request(req)
  File "D:\FindTheMag2\libs\pyboinc\_raw_client.py", line 82, in request
    return await self.receive()
  File "D:\FindTheMag2\libs\pyboinc\_raw_client.py", line 63, in receive
    buff = [await self._read()]
  File "D:\FindTheMag2\libs\pyboinc\_raw_client.py", line 57, in _read
    return await self._reader.readuntil(separator=END_OF_MESSAGE)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\asyncio\streams.py", line 613, in readuntil
    raise exceptions.IncompleteReadError(chunk, None)
asyncio.exceptions.IncompleteReadError: 0 bytes read on a total of undefined expected bytes
```

End of log file:
```
...
INFO:root:https://universeathome.pl/universe/: 1
INFO:root:https://www.worldcommunitygrid.org/boinc/: 1
INFO:root:https://www.rechenkraft.net/yoyo/: 1
DEBUG:root:Starting control of BOINC...
WARNING:root:global_prefs_override.xml does not appear to exist, not backing up. Some users may not have one. Error: [Errno 2] No such file or directory: 'C:\\ProgramData\\BOINC\\global_prefs_override.xml'
```

**Comment by makeasnek** [comment]
Fixed in latest release v 3.0
